modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -486,31 +470,26 @@ void WindowContextBase::process_key(GdkEventKey* event) {
     } else {
         jChars = mainEnv->NewCharArray(0);
     }
-    if (jview) {
-        if (press) {
-            mainEnv->CallVoidMethod(jview, jViewNotifyKey,
-                    com_sun_glass_events_KeyEvent_PRESS,
-                    glassKey,
-                    jChars,
-                    glassModifier);
-            CHECK_JNI_EXCEPTION(mainEnv)
 
-            if (jview && key > 0) { // TYPED events should only be sent for printable characters.
-                mainEnv->CallVoidMethod(jview, jViewNotifyKey,
-                        com_sun_glass_events_KeyEvent_TYPED,
-                        com_sun_glass_events_KeyEvent_VK_UNDEFINED,
-                        jChars,
-                        glassModifier);
-                CHECK_JNI_EXCEPTION(mainEnv)
-            }
-        } else {
-            mainEnv->CallVoidMethod(jview, jViewNotifyKey,
-                    com_sun_glass_events_KeyEvent_RELEASE,
-                    glassKey,
-                    jChars,
-                    glassModifier);
-            CHECK_JNI_EXCEPTION(mainEnv)
-        }
+    if (!jview) {
+        return;
+    }
+
+    mainEnv->CallVoidMethod(jview, jViewNotifyKey,
+            (press) ? com_sun_glass_events_KeyEvent_PRESS
+                    : com_sun_glass_events_KeyEvent_RELEASE,
+            glassKey,
+            jChars,
+            glassModifier);
+    CHECK_JNI_EXCEPTION(mainEnv)
+
+    if (press && key > 0) { // TYPED events should only be sent for printable characters.
```

Comment:
A handler for the PRESS event might close the window. In that case `jview` will be set to zero before you send out the TYPED event. You should do another check for that here.

See [JDK-8301219](https://bugs.openjdk.org/browse/JDK-8301219) for some sample code. I'll be submitting a fix for that bug just as soon as I get a test case working reliably.

Reply:
There's a check before.

```
    if (!jview) {
        return;
    }
```

Reply:
I see what you mean now. Fixed it.

Reply:
Looks good to me.